gbdxtools/ipe/interface.py

```diff
@@ -43,6 +52,75 @@
 
 NAMESPACE_UUID = uuid.NAMESPACE_DNS
 
+def optimize_fetch(dsk, keys=None):
+    dsk2 = {}
+    if not keys:
+        keys = dsk.keys()
+    for k in keys:
+        if dsk[k][0] is load_url:
+            dsk2[k] = (dsk[k][1], (k[1:]))
+        else:
+            dsk2[k] = dsk[k]
+    return dsk2
+
+def load_urls(collection, token, shape=(8,256,256), timeout=0.1):
+    mc = pycurl.CurlMulti()
+    nhandles = len(collection)
+    results = {}
+    handles = []
+    for url, index in collection:
+        _, ext = os.path.splitext(urlparse(url).path)
+        _curl = pycurl.Curl()
+        _curl.setopt(pycurl.NOSIGNAL, 1)
+        _curl.setopt(pycurl.HTTPHEADER, ['Authorization: Bearer {}'.format(token)])
+        _curl.setopt(pycurl.CONNECTTIMEOUT, 30)
+        _curl.setopt(pycurl.TIMEOUT, 300)
+        _curl.setopt(pycurl.URL, url)
+        _curl.fp = NamedTemporaryFile(prefix='gbdxtools', suffix=ext, delete=False)
```

Comment:
are you sure about this one?  I could be wrong but it looks like you are overriding curls internal file pointer with a file object.  I don't think fp needs to be set, the file ref just needs to be maintained.